### PR TITLE
Adds Punpun

### PR DESCRIPTION
**Pull request by Pspritechologist**
The Bartender's monkey, Punpun grows in numerance with each passing shift he manages to survive, and holds on to any hats or masks he's given for the next shift, taking ownership of them. At the fourteenth shift, he takes a well needed break and comes back refreshed.

To do is add a note claiming his vacationage, and making something more interesting happen for the fourteenth Punpun?

### CODE REVIEW

File: Content.Server/SimpleStation14/Punpun/PunpunSystem.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            metaCompItem.EntityName = EntityManager.GetComponent<MetaDataComponent>(uid).EntityName + "'s " + itemName;
+        }
+        else
+        {
+            _entityManager.DeleteEntity(itemEnt);
+        }
+    }
+
+    // Checks if an item exists in a slot, and returns its name.
+    public string checkSlot(EntityUid uid, string slot)
```
Comment (DEATHB4DEFEAT) [inline Content.Server/SimpleStation14/Punpun/PunpunSystem.cs]:
```suggestion
    private string checkSlot(EntityUid uid, string slot)
```

File: Resources/Prototypes/SimpleStation14/Entities/Objects/Misc/paper.yml [hunk 2]
```diff
@@ -19,3 +19,14 @@
...
+- type: entity
+  name: note
+  description: 'A piece of white paper.'
+  id: PaperWrittenPunpunNote
+  parent: PaperCaptainsThoughts
+  suffix: "Punpun Note"
+  components:
+  - type: Paper
+    content: |
+      I, punpun, invoke my right to have all of my clones on the NT family vacation to the meat packaging plant one out of every 15 shifts.
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/SimpleStation14/Entities/Objects/Misc/paper.yml]:
```suggestion
      I, Punpun, invoke my right to have all of my clones on the NT family vacation to the meat packaging plant one out of every 15 shifts.
```

File: Resources/Textures/SimpleStation14/Mobs/Pets/punpun.rsi/meta.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+  "version": 1,
+  "size": {
+    "x": 32,
+    "y": 32
+  },
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Taken from TGStation",
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/SimpleStation14/Mobs/Pets/punpun.rsi/meta.json]: Link to commit

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]

File: Content.Server/SimpleStation14/Punpun/PunpunSystem.cs (hunk 1 above)
Comment (DEATHB4DEFEAT) [reply Content.Server/SimpleStation14/Punpun/PunpunSystem.cs]: Capitalize function name too

File: Content.Server/SimpleStation14/Punpun/PunpunSystem.cs [hunk 4]
```diff
@@ -0,0 +1,124 @@
+using System.Linq;
+using Content.Server.GameTicking;
+using Content.Shared.Inventory;
+using Content.Shared.Mobs;
+using Content.Shared.Mobs.Components;
+
+namespace Content.Server.SimpleStation14.Punpun;
+
+public class PunpunSystem : EntitySystem
```
Comment (DEATHB4DEFEAT) [inline Content.Server/SimpleStation14/Punpun/PunpunSystem.cs]:
```suggestion
public sealed class PunpunSystem : EntitySystem
```

File: Content.Server/SimpleStation14/Punpun/PunpunSystem.cs [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        { 9, "X" },
+        { 10, "XI" },
+        { 11, "XII" },
+        { 12, "XIII" },
+        { 13, "XIV" }
+    };
+
+    public override void Initialize()
+    {
+        base.Initialize();
```
Comment (DEATHB4DEFEAT) [inline Content.Server/SimpleStation14/Punpun/PunpunSystem.cs:37]:
```suggestion
        base.Initialize();
        
```

File: Content.Server/SimpleStation14/Punpun/PunpunSystem.cs [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+    // Checks if an item exists in a slot, and returns its name.
+    private string checkSlot(EntityUid uid, string slot)
+    {
+        if (_inventorySystem.TryGetSlotEntity(uid, slot, out var item) && item != null)
+        {
+            return _entityManager.GetComponent<MetaDataComponent>(item.Value).EntityPrototype!.ID;
+        }
+        return String.Empty;
+    }
```
Comment (DEATHB4DEFEAT) [inline Content.Server/SimpleStation14/Punpun/PunpunSystem.cs]:
```suggestion
    private string CheckSlot(EntityUid uid, string slot)
    {
        if (_inventorySystem.TryGetSlotEntity(uid, slot, out var item) && item != null)
            return _entityManager.GetComponent<MetaDataComponent>(item.Value).EntityPrototype!.ID;
        
        return String.Empty;
    }
```

File: Content.Server/SimpleStation14/Punpun/PunpunSystem.cs [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+        if (_inventorySystem.TryEquip(uid, itemEnt, slot, true))
+        {
+            var metaCompItem = _entityManager.GetComponent<MetaDataComponent>(itemEnt);
+            var itemName = metaCompItem.EntityName;
+            metaCompItem.EntityName = EntityManager.GetComponent<MetaDataComponent>(uid).EntityName + "'s " + itemName;
+        }
+        else
+        {
+            _entityManager.DeleteEntity(itemEnt);
+        }
```
Comment (DEATHB4DEFEAT) [inline Content.Server/SimpleStation14/Punpun/PunpunSystem.cs]:
```suggestion
        else _entityManager.DeleteEntity(itemEnt);
```

File: Content.Server/SimpleStation14/Punpun/PunpunSystem.cs [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+            punpunData = (0, String.Empty, String.Empty);
+            return;
+        }
+
+        punpunData.Item1++;
+
+        if (_entityManager.TryGetComponent<InventoryComponent>(uid, out var invComp))
+        {
+            punpunData.Item2 = checkSlot(uid, "head");
+            punpunData.Item3 = checkSlot(uid, "mask");
```
Comment (DEATHB4DEFEAT) [inline Content.Server/SimpleStation14/Punpun/PunpunSystem.cs]:
```suggestion
            punpunData.Item2 = CheckSlot(uid, "head");
            punpunData.Item3 = CheckSlot(uid, "mask");
```

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]

File: Content.Server/SimpleStation14/Punpun/PunpunSystem.cs [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+    private void OnRoundStart(EntityUid uid, PunpunComponent component, ComponentStartup args)
+    {
+        component.owner = uid;
+
+        if (punpunData.Item1 > 13)
+        {
+            _entityManager.SpawnEntity("PaperWrittenPunpunNote", EntityManager.GetComponent<TransformComponent>(uid).Coordinates);
+            _entityManager.QueueDeleteEntity(uid);
+            punpunData = (0, String.Empty, String.Empty);
+            return;
```
Comment (DEATHB4DEFEAT) [inline Content.Server/SimpleStation14/Punpun/PunpunSystem.cs:54]:
```suggestion

            return;
```

File: Content.Server/SimpleStation14/Punpun/PunpunSystem.cs [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+        }
+
+        var metaCompEnt = _entityManager.GetComponent<MetaDataComponent>(uid);
+        var name = metaCompEnt.EntityName;
+        metaCompEnt.EntityName = name + " " + numerals[punpunData.Item1];
```
Comment (DEATHB4DEFEAT) [inline Content.Server/SimpleStation14/Punpun/PunpunSystem.cs:60]:
```suggestion

        metaCompEnt.EntityName = name + " " + numerals[punpunData.Item1];
```

File: Content.Server/SimpleStation14/Punpun/PunpunSystem.cs [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+    {
+        if (item == String.Empty)
+            return;
+
+        var itemEnt = EntityManager.SpawnEntity(item, EntityManager.GetComponent<TransformComponent>(uid).Coordinates);
+        if (_inventorySystem.TryEquip(uid, itemEnt, slot, true))
+        {
+            var metaCompItem = _entityManager.GetComponent<MetaDataComponent>(itemEnt);
+            var itemName = metaCompItem.EntityName;
+            metaCompItem.EntityName = EntityManager.GetComponent<MetaDataComponent>(uid).EntityName + "'s " + itemName;
```
Comment (DEATHB4DEFEAT) [inline Content.Server/SimpleStation14/Punpun/PunpunSystem.cs:111]:
```suggestion

            metaCompItem.EntityName = EntityManager.GetComponent<MetaDataComponent>(uid).EntityName + "'s " + itemName;
```

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]
nitpicking :)

**Review by DEATHB4DEFEAT — APPROVED** [review]
lgtm

**Comment by DEATHB4DEFEAT** [comment]
looks like you had further plans, merge whenever you want

**Comment by Pspritechologist** [comment]
I don't think I did? Believe this one is done

**Comment by DEATHB4DEFEAT** [comment]
what's up with the changed files?

**Comment by Pspritechologist** [comment]
> what's up with the changed files?

Pulling issues. I'll rebase this, that should fix it.

**Comment by Pspritechologist** [comment]
@DEATHB4DEFEAT :D

**Review by DEATHB4DEFEAT — APPROVED** [review]

**Pull request merged**